### PR TITLE
[Td - 2179 ] Rich Message video template Feature

### PR DESCRIPTION
**Pull request by Sathyan-Elangovan**
1. Added the Rich Messages Video Template Feature.

We have two types of payloads to play videos in our sdk.
1. Payload for web hosted videos
2. Payload for Youtube videos

Below is the sample custom payload in our documentation.
[Doucmentation link for Rich Video Template](https://docs.kommunicate.io/docs/message-types#videos--youtube
)
```
{
  "platform": "kommunicate",
  "message": "Hello this is example videos",
  "metadata": {
    "contentType": "300",
    "templateId": "14",
    "payload": [
      {
        "url": "https://s3-migration-dot-applozic.appspot.com/rest/ws/aws/file/AM
Ifv95BPKLnhNtiYz0SNJYXTDhUkvhPYQikSjFaoL1hbg1zCcK8n6KZZfZC5Z42bfxOLaNeKSbKrt4o9pI0_ICb1fKKY8HqtcCeEuUZU8b18u8sr-4U5d4rFtmC79cn_0xaiV0MQfF01T0hAea8sQANoaYuvR5hQJHiFSB047yp1mkfqcwNb_ExaX1xutdJ8Qamkjp8xTKjlPiG0920T7qqLFIBxl1eGL7ikvTCtXG_f8Me8OKsEOhUXOP7LLuJLcOG_p8cijsp",
        "caption": "Video & Youtube Examples"
      },
      {
        "source": "youtube",
        "url": "https://www.youtube.com/embed/3-ijFzzF934"
      }
    ]
  }
}
```
**Sample Images:**

**Sample Video:**

https://user-images.githubusercontent.com/61688116/144352056-418bd8a1-7ad5-41d7-b971-610af4685d39.mp4

### CODE REVIEW

**Review by amntoppo — APPROVED** [review]

**Comment by devashishmamgain** [comment]
@Sathyan-Elangovan what is the size of the new library?
How much SDK size will increase after this inclusion?

**Comment by amntoppo** [comment]
> > @Sathyan-Elangovan what is the size of the new library?
> > How much SDK size will increase after this inclusion?
> 
> @devashishmamgain Downloadable size is increased by 2.7 MB

ExoPlayer libraries modules can be added as dependency separately. You can have a look, if we are not using the whole library then we can add the modules separately. That should decrease the app size.

**Review by devashishmamgain — DISMISSED** [review]

File: kommunicateui/build.gradle [hunk 1]
```diff
@@ -47,6 +50,9 @@ dependencies {
     api "androidx.swiperefreshlayout:swiperefreshlayout:1.1.0"
     implementation 'com.google.android.flexbox:flexbox:3.0.0'
     implementation 'androidx.constraintlayout:constraintlayout:2.0.4'
+
+    implementation 'com.google.android.exoplayer:exoplayer:2.15.0'
```
[the remark this answers is not in the document]
Comment (Sathyan-Elangovan) [reply kommunicateui/build.gradle]: @devashishmamgain @adarshmishra @mukeshthawani  @amntoppo  Found an another solution that will have same sdk size. There will no increase in sdk size due to this feature.

File: kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/richmessaging/adapters/KmVideoRMAdapter.java [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+            webSettings.setJavaScriptEnabled(true);
+            holder.webview.setWebChromeClient(new WebChromeClient());
+
+            holder.webview.getSettings().setPluginState(WebSettings.PluginState.ON);
+            if (!TextUtils.isEmpty(payloadModel.getSource())) {
+                holder.videoViewRoot.setVisibility(View.GONE);
+                holder.captionText.setVisibility(View.GONE);
+                holder.webViewRoot.setVisibility(View.VISIBLE);
+                String currentUrl = "<iframe width=\"100%\" height=\"100%\" src=\"" + payloadModel.getUrl() + "\" frameborder=\"0\" allow=\"accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture\" allowfullscreen></iframe>";
+                holder.webview.loadData(currentUrl, "text/html", "utf-8");
```
Comment (adarshmishra) [inline kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/richmessaging/adapters/KmVideoRMAdapter.java:84]: Will this always open the webview (any browser), any particualr reason for not opening the default app and webview?

Comment (Sathyan-Elangovan) [reply kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/richmessaging/adapters/KmVideoRMAdapter.java:84]: its not external webview. This webview xml component inside Recyclerview item layout.

File: kommunicateui/src/main/res/layout/km_rich_message_video_item.xml [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        android:layout_width="match_parent"
+        android:layout_height="200dp">
+
+        <WebView
+            android:id="@+id/web_view"
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            android:background="@color/black"
+            android:scrollbars="none" />
+    </FrameLayout>
```
Comment (Sathyan-Elangovan) [inline kommunicateui/src/main/res/layout/km_rich_message_video_item.xml:108]: @adarshmishra you tube url will be played on this webview.

**Review by adarshmishra — APPROVED** [review]

**Pull request merged**